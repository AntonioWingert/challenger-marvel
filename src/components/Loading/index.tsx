import * as S from './styles';

export default function Loading() {
  return (
    <S.Container>
      <S.LoadingSpinner aria-label='loading-spinner'/>
    </S.Container>
  );
}