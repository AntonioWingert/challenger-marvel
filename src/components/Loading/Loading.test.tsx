import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import Loading from '.';

describe('<Loading />', () => {
  it('should render correctly with default props', () => {
    const { container } = render(
      <Loading />
    );

    expect(screen.getByLabelText(/loading-spinner/i)).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });
});