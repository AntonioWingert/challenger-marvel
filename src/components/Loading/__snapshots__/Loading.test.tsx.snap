// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Loading /> > should render correctly with default props 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
}

.c1 {
  width: 50px;
  height: 50px;
  border: 10px solid #f3f3f3;
  border-top: 10px solid #383636;
  border-radius: 50%;
  animation: spinner 1.5s linear infinite;
}

<div
  class="c0"
>
  <div
    aria-label="loading-spinner"
    class="c1"
  />
</div>
`;
